### ISSUE TITLE
Bodhi puts too much text into the Greenwave summary string

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#2334]
A recent change (#2273) expanded the amount of text that would be stored in Greenwave's summary string. Unfortunately, updates now hit the limit of the size of the ```greenwave_summary_string``` column in the database now, leading to failures in the ```bodhi-check-policies``` cron job:

```
[root@bodhi-backend01 ~][STG]# time bodhi-check-policies                                                              
(psycopg2.DataError) value too long for type character varying(255)                                                   
 [SQL: 'UPDATE updates SET greenwave_summary_string=%(greenwave_summary_string)s WHERE updates.id = %(updates_id)s'] [parameters: {'updates_id': 111436, 'greenwave_summary_string': u'18 of 35 required test results missing\n Missing: dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdepli ... (57 characters truncated) ... st.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint'}]
(psycopg2.DataError) value too long for type character varying(255)                                                   
 [SQL: 'UPDATE updates SET greenwave_summary_string=%(greenwave_summary_string)s WHERE updates.id = %(updates_id)s'] [parameters: {'updates_id': 111529, 'greenwave_summary_string': u'18 of 70 required test results missing\n Missing: dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdepli ... (57 characters truncated) ... st.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint, dist.rpmdeplint'}]
```

The data in this case is redundant. I think we should make the code that creates this string a little smarter in a couple ways:

* It should not repeat the same test name again and again, but should instead print the name once and perhaps the number of occurrences.
* It should verify that the final string it produces does not exceed 255 characters. It could finish with elipses if there were more failures than fit in 255 characters.

In any case, this is a 3.7.0 blocker as it is a regression from 3.6.1.

**Comment by pypingou** [comment]
For the second point, what about changing this field from varchar(255) to text()?

**Comment by bowlofeggs** [comment]
@pypingou Since the string is called a "summary" I don't think we should need to make it an unbounded text field. I thought about this problem some over the weekend, and I think the right solution is to make a dedicated field for this data, perhaps even using PostgreSQL's json type to store it. This way the summary can remain the summary, and no special demarcation string is needed to separate the summary from the missing tests.

I think this same patch can fix #2334 and #2335.
